### PR TITLE
feat: Start and finish renting

### PR DESCRIPTION
**Pull request by marijakljestan**

### CODE REVIEW

File: backend/src/main/java/com/example/vehiclesservice/service/RentingServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+@Service
+@AllArgsConstructor
+public class RentingServiceImpl implements RentingService{
+
+    private final RentingRepository rentingRepository;
+    private final VehicleService vehicleService;
+    private final ModelMapper modelMapper = new ModelMapper();
+
+    @Override
+    public Renting startRenting(Renting renting) throws VehicleNotFoundException {
```
Comment (ilicjelena) [inline backend/src/main/java/com/example/vehiclesservice/service/RentingServiceImpl.java:25]: Think what will happen if two users want to reserve the same vehicle at the same time

File: backend/src/main/java/com/example/vehiclesservice/api/Address.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package com.example.vehiclesservice.api;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data @NoArgsConstructor@AllArgsConstructor@Builder
```
Comment (mjevtovic9) [inline backend/src/main/java/com/example/vehiclesservice/api/Address.java:8]: It's more readable when these are in multiple lines

File: backend/src/main/java/com/example/vehiclesservice/api/Vehicle.java [hunk 3]
```diff
@@ -13,7 +13,8 @@
     private String model;
     private String description;
     private double pricePerDay;
-    private Garage garage;
+    private GarageEntity garage;
```
Comment (mjevtovic9) [inline backend/src/main/java/com/example/vehiclesservice/api/Vehicle.java:16]: Entity cannot be in the api package

File: backend/src/main/java/com/example/vehiclesservice/config/ModelMapperConfig.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Configuration
+public class ModelMapperConfig {
+    @Bean
+    public ModelMapper modelMapper() {
```
Comment (mjevtovic9) [inline backend/src/main/java/com/example/vehiclesservice/config/ModelMapperConfig.java:13]: You can autowire this bean in `GarageServiceImpl`, `RentingServiceImpl`, `VehicleServiceImpl` without creating a new mapper in each of these services.

File: backend/src/main/java/com/example/vehiclesservice/service/GarageServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    private final GarageRepository garageRepository;
+    private final ModelMapper modelMapper = new ModelMapper();
+
+    @Override
+    public List<Garage> getAllGarages() {
+        List<GarageEntity> garages = garageRepository.findAll();
+        return mapList(garages, Garage.class);
+    }
+
+    private <S, T> List<T> mapList(List<S> source, Class<T> targetClass) {
```
Comment (mjevtovic9) [inline backend/src/main/java/com/example/vehiclesservice/service/GarageServiceImpl.java:26]: Maybe this method can be extracted as a static method somewhere else so it doesn't repeat across services.

File: backend/src/main/java/com/example/vehiclesservice/api/Address.java (hunk 2 above)
Comment (mzorboski) [reply backend/src/main/java/com/example/vehiclesservice/api/Address.java:8]: Also, are both constructors needed?

File: backend/src/main/java/com/example/vehiclesservice/controller/GaragesController.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping(value = "/garage", produces = MediaType.APPLICATION_JSON_VALUE)
+@AllArgsConstructor
+public class GaragesController {
+
+    private final GarageService garageService;
+
+    @GetMapping("/getAll")
```
Comment (jelenailic95) [inline backend/src/main/java/com/example/vehiclesservice/controller/GaragesController.java:18]: You should use just /garages for getAll.
Not necessary to change, just to mention some convention :)
GET       /garages - get all garages
GET       /garages/{id} - get garage by id 
POST     /garages - add new garage
PUT       /garages/{id} - update garage
DELETE  /garages/{id} - delete garage ...

File: backend/src/main/java/com/example/vehiclesservice/exception/RentingNotFoundException.java [hunk 7]
```diff
@@ -0,0 +1,7 @@
+package com.example.vehiclesservice.exception;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(code = HttpStatus.NOT_FOUND, reason = "Vehicle not found!")
+public class RentingNotFoundException extends NullPointerException{}
```
Comment (jelenailic95) [inline backend/src/main/java/com/example/vehiclesservice/exception/RentingNotFoundException.java:7]: Custom exceptions should extends RuntimeException

**Review by mjevtovic9 — APPROVED** [review]

**Pull request merged**